src/rrdp.rs

```diff
@@ -680,15 +688,43 @@ impl CurrentObject {
         &self.data
     }
 
-    pub fn hash(&self) -> Hash {
-        Hash::from_data(&self.data)
+    pub fn since(&self) -> Time {
+        self.since
+    }
+
+    /// Fixes the since time based on the actual parsed object.
+    /// Returns an error if the object cannot be parsed.
+    fn fix_since(&mut self) -> Result<()> {
+        let uri_path= self.uri.path();
+        if uri_path.ends_with(".cer") {
+            let cer = Cert::decode(self.data.as_ref())
+                .map_err(|_| anyhow!("Cannot parse certificate"))?;
+            
+            self.since = cer.validity().not_before().into();
+        } else if uri_path.ends_with(".mft") {
+            let mft = Manifest::decode(self.data.as_ref(), false)
+                .map_err(|_| anyhow!("Cannot parse manifest"))?;
+            
+            self.since = mft.this_update().into();
```

Comment:
How about gbr (and is there a generic way to handle CMS?)

Reply:
Added a generic RPKI signed object parse attempt.